libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -113,8 +113,17 @@ func AutodiscoverBuilder(bus bus.Bus, uuid uuid.UUID, c *common.Config) (autodis
 		},
 		UpdateFunc: func(obj kubernetes.Resource) {
 			logp.Debug("kubernetes", "Watcher Pod update: %+v", obj)
-			p.emit(obj.(*kubernetes.Pod), "stop")
-			p.emit(obj.(*kubernetes.Pod), "start")
+			pod, _ := obj.(*kubernetes.Pod)
+
+			// Avoid restarting a module when update events for a
+			// terminated pod are received
+			if podTerminated(pod) {
+				logp.Debug("kubernetes", "Pod terminated, stopping instead of restarting on update")
```

Comment:
Not sure, but I think that in deployments where it happens it will happen with frequency.

I am also not sure if the overlap between delete and update events is caused by our implementation or if kubernetes is really sending an update after a delete. This will need more investigation. This code is more defensive that a real fix, I think that no update should arrive or be processed after a delete, not sure why this is happening.

I'd leave this message as debug, the only reason I added it was to know if this path was being executed in case we have to debug it in the future.